pkg/helm/kube.go

```diff
@@ -33,27 +33,38 @@ var kubeClient *kube
 var once sync.Once
 
 // GetConfigInstance returns a Kubernetes interface based on the current configuration
-func getConfigInstance() *kube {
+func getConfigInstance(kubeContext string) (*kube, error) {
+	var err error
+	var client kubernetes.Interface
+
 	once.Do(func() {
 		if kubeClient == nil {
+			client, err = getKubeClient(kubeContext)
+
 			kubeClient = &kube{
-				Client: getKubeClient(),
+				Client: client,
 			}
 		}
 	})
-	return kubeClient
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Should this go inside the once.Do() func? I don't think it's possible to have an error after the above function works once especially since we instantiate the err variable above as a nil error every time `getConfigInstance` is run.

Reply:
This isn't blocking as it will still work as it is, just curious.

Reply:
I think it's working, and covered by the test? We can't return the error inside of once because that's an internal function

Reply:
Ah right, that makes sense. Not enough coffee this morning.